runtime/src/accounts_db.rs

```diff
@@ -124,6 +129,97 @@ impl Versioned for (u64, AccountInfo) {
     }
 }
 
+pub enum LoadedAccountAccessor<'a> {
+    Stored(Option<(Arc<AccountStorageEntry>, usize)>),
+    Cached((&'a AccountsCache, Slot, &'a Pubkey)),
+}
+
+impl<'a> LoadedAccountAccessor<'a> {
+    fn get_loaded_account(&self) -> Option<LoadedAccount> {
+        match self {
+            LoadedAccountAccessor::Stored(storage_entry) => {
+                // May not be present if slot was cleaned up in between
+                storage_entry.as_ref().and_then(|(storage_entry, offset)| {
+                    storage_entry
+                        .get_stored_account_meta(*offset)
+                        .map(LoadedAccount::Stored)
+                })
+            }
+            LoadedAccountAccessor::Cached((cache, slot, pubkey)) => {
+                // May not be present if slot was cleaned up in between
+                cache.load(*slot, pubkey).map(|cached_account| {
+                    LoadedAccount::Cached((**pubkey, Cow::Owned(cached_account)))
+                })
+            }
+        }
+    }
+}
+
+pub enum LoadedAccount<'a> {
+    Stored(StoredAccountMeta<'a>),
+    Cached((Pubkey, Cow<'a, CachedAccount>)),
+}
+
+impl<'a> LoadedAccount<'a> {
+    pub fn owner(&self) -> &Pubkey {
+        match self {
+            LoadedAccount::Stored(stored_account_meta) => &stored_account_meta.account_meta.owner,
+            LoadedAccount::Cached((_, cached_account)) => &cached_account.account.owner,
+        }
+    }
+
+    pub fn executable(&self) -> bool {
+        match self {
+            LoadedAccount::Stored(stored_account_meta) => {
+                stored_account_meta.account_meta.executable
+            }
+            LoadedAccount::Cached((_, cached_account)) => cached_account.account.executable,
+        }
+    }
+
+    pub fn stored_hash(&self) -> &Hash {
```

Comment:
nits: I think this fn should be renamed to `loaded_hash`? I sensed you're consistently using for loaded for AppendVec/Cache and stored for AppendVec.

Reply:
yup, made it more consistent, thanks!